client/extensions/woocommerce/woocommerce-services/state/carrier-accounts/selectors.js

```diff
@@ -39,17 +43,38 @@ export const getFormErrors = ( state, siteId, carrier ) => {
 		}
 	}
 
+	if ( values.account_number && ! values.account_number.match( UPSAccountNumberRegex ) ) {
+		fieldErrors.account_number = translate( 'The UPS account number needs to be 6 digits in length' );
```

Comment:
The account number can be digits or letters, its length should be max 6 characters.
This is validating only on the digits, it should also include `[a-zA-Z]`

Reply:
Got it, @frosso, I didn't realise this! I was following @elizaan36's note on the linked issue blindly, like a giant worm chasing Kevin Bacon off the edge of a cliff. I've just updated this PR with a new commit that corrects this. You should now be able to see the following error messages: 

![Error state](https://cdn-std.droplr.net/files/acc_1104206/f9Z6wL)
_Error state_

![Letters accepted](https://cdn-std.droplr.net/files/acc_1104206/iUBUku)
_Letters accepted_

```
const UPSAccountNumberRegex = /^[a-zA-Z0-9]{6}$/;
```
_New regex_

Also, I've today realised that my GitHub notifications are basically all set wrong, so sorry @frosso for taking so long to notice your feedback! 🤦

Reply:
haha no worries! :) The new regex looks good!